components/result/run-history_test.go

```diff
@@ -0,0 +1,413 @@
+package result
+
+import (
+	"testing"
+	"time"
+
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	"github.com/litmuschaos/litmus-e2e/pkg"
+	"github.com/litmuschaos/litmus-e2e/pkg/environment"
+	"github.com/litmuschaos/litmus-e2e/pkg/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/klog"
+)
+
+func TestRunHistory(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+// BDD for testing components
+var _ = Describe("BDD of pod-delete experiment", func() {
+
+	// BDD TEST CASE 1
+	Context("Run history check for passed run", func() {
+
+		It("Should check for the run history in the chaos result when experiment is passed", func() {
+
+			testsDetails := types.TestDetails{}
+			clients := environment.ClientSets{}
+			chaosExperiment := v1alpha1.ChaosExperiment{}
+			chaosEngine := v1alpha1.ChaosEngine{}
+
+			// Getting kubeConfig and Generate ClientSets
+			By("[PreChaos]: Getting kubeconfig and generating Clientsets")
+			err := clients.GenerateClientSetFromKubeConfig()
+			Expect(err).To(BeNil(), "Unable to get the kubeconfig, due to {%v}", err)
+
+			// Fetching all the default ENV
+			By("[PreChaos]: Fetching all default ENVs")
+			klog.Infof("[PreReq]: Getting the ENVs for the %v test", testsDetails.ExperimentName)
+			environment.GetENV(&testsDetails, "pod-delete", "run-history-passed1")
+			// environment.GetENV(&testsDetails, "pod-delete", "run-history")
```

Comment:
Can we delete this line if this is not required

Reply:
I have commented this line because the engine was not being re-applied, so I was not able to run the tests with same engine name to test consecutive run history. I have mentioned this in the initial comment